ecr-login/ecr.go

```diff
@@ -52,13 +53,21 @@ func (self ECRHelper) Get(serverURL string) (string, string, error) {
 		return "", "", credentials.NewErrCredentialsNotFound()
 	}
 
+	profile, profile_exists := os.LookupEnv("AWS_PROFILE")
+
 	var client api.Client
 	if registry.FIPS {
 		client, err = self.ClientFactory.NewClientWithFipsEndpoint(registry.Region)
 		if err != nil {
 			logrus.WithError(err).Error("Error resolving FIPS endpoint")
 			return "", "", credentials.NewErrCredentialsNotFound()
 		}
+	} else if profile_exists {
+		client, err = self.ClientFactory.NewClientWithExplicitProfile(profile)
+		if err != nil {
+            logrus.WithError(err).Error("Error creating client with explicit profile")
+            return "", "", credentials.NewErrCredentialsNotFound()
+        }
```

Comment:
This is probably a question for @samuelkarp - Right now FIPS and AWS_PROFILE are mutually-exclusive. Is there a case where a customer want to use AWS_PROFILE, but with FIPS?

Reply:
Yes, these options should not be mutually exclusive.